src/averageAges.js

```diff
@@ -55,7 +69,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people
+    .filter(child => people
+      .find(mother => mother.name === child.mother && (
+        !onlyWithSon || child.sex === 'm')));
+
+  const ageDifference = children.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  const totalAge = ageDifference
+    .reduce((prev, difference) => prev + difference, 0);
+
+  const averageAge = totalAge / ageDifference.length;
+
+  return averageAge;
```

Comment:
```suggestion
  return totalAge / ageDifference.length;;
```
not needed to create additional variable here :)